### PR TITLE
Add example to be more clear about meaning of inputs

### PR DESCRIPTION
**Pull request by KronosTheLate**
This PR adds an equation showing the form of the problem being solved for eigen(A) and eigen(A, B).
This is particularily useful to show how the arguments A and B are interpreted for 2 input matrices, and added to the single-input case for completeness and consistency.

### CODE REVIEW

File: stdlib/LinearAlgebra/src/eigen.jl [hunk 1]
```diff
@@ -480,6 +482,8 @@ end
 Compute the generalized eigenvalue decomposition of `A` and `B`, returning a
 [`GeneralizedEigen`](@ref) factorization object `F` which contains the generalized eigenvalues in
 `F.values` and the generalized eigenvectors in the columns of the matrix `F.vectors`.
+This corresponds to solving a generalized eigenvalue problem of the form
+`Ax =  λMx`, where `A, B` are matrices, `x` is an eigenvector, and  `λ` is an eigenvalue.
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/eigen.jl:486]:
```suggestion
`Ax =  λBx`, where `A, B` are matrices, `x` is an eigenvector, and  `λ` is an eigenvalue.
```

File: stdlib/LinearAlgebra/src/eigen.jl [hunk 2]
```diff
@@ -182,7 +182,9 @@ end
 
 Compute the eigenvalue decomposition of `A`, returning an [`Eigen`](@ref) factorization object `F`
 which contains the eigenvalues in `F.values` and the eigenvectors in the columns of the
-matrix `F.vectors`. (The `k`th eigenvector can be obtained from the slice `F.vectors[:, k]`.)
+matrix `F.vectors`. This corresponds to solving a generalized eigenvalue problem of the form
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/eigen.jl:185]:
```suggestion
matrix `F.vectors`. This corresponds to solving a eigenvalue problem of the form
```

File: stdlib/LinearAlgebra/src/eigen.jl (hunk 1 above)
Comment (KronosTheLate) [reply stdlib/LinearAlgebra/src/eigen.jl:486]: Right you are, thnx

File: stdlib/LinearAlgebra/src/eigen.jl [hunk 3]
```diff
@@ -182,7 +182,9 @@ end
 
 Compute the eigenvalue decomposition of `A`, returning an [`Eigen`](@ref) factorization object `F`
 which contains the eigenvalues in `F.values` and the eigenvectors in the columns of the
-matrix `F.vectors`. (The `k`th eigenvector can be obtained from the slice `F.vectors[:, k]`.)
+matrix `F.vectors`. This corresponds to solving a eigenvalue problem of the form
+`Ax =  λx`, where `A` is a matrix, `x` is an eigenvector, and  `λ` is an eigenvalue.
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/eigen.jl:186]:
```suggestion
matrix `F.vectors`. This corresponds to solving an eigenvalue problem of the form
`Ax =  λx`, where `A` is a matrix, `x` is an eigenvector, and `λ` is an eigenvalue.
```

File: stdlib/LinearAlgebra/src/eigen.jl [hunk 4]
```diff
@@ -480,6 +482,8 @@ end
 Compute the generalized eigenvalue decomposition of `A` and `B`, returning a
 [`GeneralizedEigen`](@ref) factorization object `F` which contains the generalized eigenvalues in
 `F.values` and the generalized eigenvectors in the columns of the matrix `F.vectors`.
+This corresponds to solving a generalized eigenvalue problem of the form
+`Ax =  λBx`, where `A, B` are matrices, `x` is an eigenvector, and  `λ` is an eigenvalue.
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/eigen.jl:486]:
```suggestion
`Ax =  λBx`, where `A, B` are matrices, `x` is an eigenvector, and `λ` is an eigenvalue.
```

**Review by dkarrasch — COMMENTED** [review]
Found two more details.

**Comment by KronosTheLate** [comment]
> Found two more details.

Wow, you really have a sharp eye for this. Thanks!

**Pull request merged**